src/jvm.jl

```diff
@@ -16,17 +16,47 @@ const JNI_ENOMEM       = convert(Cint, -4)              #/* not enough memory */
 const JNI_EEXIST       = convert(Cint, -5)              #/* VM already created */
 const JNI_EINVAL       = convert(Cint, -6)              #/* invalid arguments */
 
+function javahome_winreg()
+    processor_architecture = ENV["PROCESSOR_ARCHITECTURE"]
+    keypath = utf16("")
+    javahome = ""
+    cv = "CurrentVersion"
+    jh = "JavaHome"
+    global const msvclib = "msvcr100.dll"
+    
+    if processor_architecture == "AMD64" 
+        if WORD_SIZE == 64
+            keypath = utf16("SOFTWARE\\JavaSoft\\Java\ Development\ Kit")
+        elseif WORD_SIZE == 32
+            keypath = utf16("SOFTWARE\\WoW6432Node\\JavaSoft\\Java\ Development\ Kit")
```

Comment:
From some quick googling, it seems as though we shouldn't need to do this e.g. see https://msdn.microsoft.com/en-us/library/windows/desktop/aa384232(v=vs.85).aspx